xapian-letor/ranker/listnet_ranker.cc

```diff
@@ -173,7 +175,9 @@ ListNETRanker::load_model_from_metadata(const string & model_key) {
     istringstream model_str(model_data);
     string line;
     while (getline(model_str, line)) {
-	double parameter = stod(line);
+	const char *ptr = line.data();
+	const char *end = ptr + line.size();
+	double parameter = unserialise_double(&ptr, end);
 	loaded_parameters.push_back(parameter);
     }
```

Comment:
And here, no need to read line-by-line or use `istringstream` at all - just unserialise directly from `model_str`:

```
const char *ptr = model_str.data();
const char *end = ptr + model_str.size();
while (ptr != end)
    loaded_parameters.push_back(unserialise_double(&ptr, end));
```